Source/Private/Source.cpp

```diff
@@ -71,10 +93,76 @@ bool CheckMap(BYTE* fileBuffer, int bufferLen, noeRAPI_t* rapi)
 	return 1;
 }
 
+template<bool bBigEndian>
+bool CheckDZArchive(BYTE* fileBuffer, int bufferLen, noeRAPI_t* rapi)
+{
+	uint8_t ed = *(uint8_t*)(fileBuffer);
+	return !(bBigEndian ^ (ed == 0x0 ? true : false));
```

Comment:
This is hard to mentally unravel and can be simplified:

```
return !(bBigEndian ^ (ed == 0x0 ? true : false));
return !(bBigEndian ^ (ed != 0x0));
return (bBigEndian ^ (ed == 0x0));
return bBigEndian ^ (ed == 0x0);
return bBigEndian ^ !ed; // If you want really concise 😉
```

```suggestion
	return bBigEndian ^ (ed == 0x0);
```